openproblems/tasks/_cell_cell_communication/_common/metrics/prauc.py

```diff
@@ -0,0 +1,51 @@
+from .....tools.decorators import metric
+from sklearn.metrics import auc
+from sklearn.metrics import precision_recall_curve
+
+import numpy as np
+
+
+@metric(metric_name="Precision-recall AUC", maximize=True)
+def prauc(adata, merge_keys, n_times=100):
+    gt = adata.uns["ccc_target"].merge(
+        adata.uns["ccc_pred"], on=merge_keys, how="outer"
+    )
+
+    gt.loc[gt["response"].isna(), "response"] = 0
+    gt.loc[gt["score"].isna(), "score"] = np.nanmin(gt["score"]) - 1
+
+    cpn = np.sum(gt["response"] == 1)  # condition positive num
+    cnn = np.sum(gt["response"] == 0)  # condition negative num
+
+    if (cpn != cnn) & (n_times > 1):
```

Comment:
Why not just use

```
gt["response"] -= gt["response"].min()
gt["response"] /= gt["response"].max()
precision, recall, _ = sklearn.metrics.precision_recall_curve(gt["score"], gt["response"])
auc = sklearn.metrics.auc(recall, precision)
```
?

Reply:
This is no longer relevant as we agreed to remove the downsampling :)